ldelf/ta_elf_rel.c

```diff
@@ -191,27 +197,30 @@ static void e32_relocate(struct ta_elf *elf, unsigned int rel_sidx)
 	}
 
 	/* Check the address is inside TA memory */
-	assert(shdr[rel_sidx].sh_addr < (elf->max_addr - elf->load_addr));
+	if (ADD_OVERFLOW(shdr[rel_sidx].sh_addr,
```

Comment:
Prior this change, `shdr[rel_sidx].sh_addr` and end address were asserted against max end address. With this change, only end address is checked. Is `shdr[rel_sidx].sh_addr` checked before? (I can't find where)

Reply:
`shdr[rel_sidx].sh_addr` cannot be less than `0` so the addition and check of `sh_end` should cover `shdr[rel_sidx].sh_addr` too.